akka-distributed-data/src/main/scala/akka/cluster/ddata/ORMap.scala

```diff
@@ -125,11 +214,42 @@ final class ORMap[A, B <: ReplicatedData] private[akka] (
    * INTERNAL API
    */
   @InternalApi private[akka] def updated(node: UniqueAddress, key: A, initial: B)(modify: B ⇒ B): ORMap[A, B] = {
-    val newValue = values.get(key) match {
-      case Some(old) ⇒ modify(old)
-      case _         ⇒ modify(initial)
+    val (oldValue, hasOldValue) = values.get(key) match {
```

Comment:
`updated` with no previous value is a `put`. This will be indicated by `hasOldValue`